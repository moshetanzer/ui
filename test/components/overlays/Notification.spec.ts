import { describe, it, expect } from 'vitest'
import { UNotification } from '#components'
import type { TypeOf } from 'zod'
import ComponentRender from '../component-render'

describe('Notification', () => {
  it.each([
    ['with title', { props: { title: 'Notification title' } }],
    ['with description', { props: { description: 'Notification description' } }],
    ['with icon', { props: { icon: 'i-heroicons-academic-cap' } }],
    ['with avatar', { props: { avatar: { src: 'https://repository-images.githubusercontent.com/428329515/43fec891-9030-4601-8233-5d45ba5c6013', alt: 'nuxtui' } } }],
    ['with close button', { props: { closeButton: { label: 'close', color: 'red' } } }],
    ['with color', { props: { color: 'red' } }]
    // // @ts-ignore
  ])('renders %s correctly', async (nameOrHtml: string, options: TypeOf<typeof UNotification.props>) => {
    let html = await ComponentRender(nameOrHtml, options, UNotification)
    // avoid test failure due to width variations
    html = html.replace('style="width: 100%;"', '')
    html = html.replace('style="width: 100.02%;"', '')
    expect(html).toMatchSnapshot()
  })
})
