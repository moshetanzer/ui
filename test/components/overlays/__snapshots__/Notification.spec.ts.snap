// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Notification > renders with avatar correctly 1`] = `
"<transition-stub enteractiveclass="transform ease-out duration-300 transition" enterfromclass="translate-y-2 opacity-0 sm:translate-y-0 sm:translate-x-2" entertoclass="translate-y-0 opacity-100 sm:translate-x-0" leaveactiveclass="transition ease-in duration-100" leavefromclass="opacity-100" leavetoclass="opacity-0" appear="true" persisted="false" css="true">
  <div class="w-full pointer-events-auto bg-white dark:bg-gray-900 rounded-lg shadow-lg" role="status">
    <div class="relative overflow-hidden rounded-lg ring-1 ring-gray-200 dark:ring-gray-800">
      <div class="flex p-4 gap-3 items-center">
        <!--v-if--><span class="relative inline-flex items-center justify-center flex-shrink-0 rounded-full h-10 w-10 text-base flex-shrink-0 self-center"><img class="rounded-full h-10 w-10 text-base" alt="nuxtui" src="https://repository-images.githubusercontent.com/428329515/43fec891-9030-4601-8233-5d45ba5c6013"><!--v-if--></span>
        <div class="w-0 flex-1">
          <!--v-if-->
          <!--v-if-->
          <!--v-if-->
        </div>
        <div class="flex items-center gap-2 flex-shrink-0 mt-0">
          <!--v-if--><button type="button" class="focus:outline-none focus-visible:outline-0 disabled:cursor-not-allowed disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 text-gray-500 dark:text-gray-400 hover:text-gray-700 dark:hover:text-gray-200 underline-offset-4 hover:underline focus-visible:ring-inset focus-visible:ring-2 focus-visible:ring-primary-500 dark:focus-visible:ring-primary-400 inline-flex items-center" aria-label="Close"><span class="i-heroicons-x-mark-20-solid flex-shrink-0 h-5 w-5" aria-hidden="true"></span>
            <!--v-if-->
            <!--v-if-->
          </button>
        </div>
      </div>
      <div class="absolute bottom-0 end-0 start-0 h-1 bg-primary-500 dark:bg-primary-400" ></div>
    </div>
  </div>
</transition-stub>"
`;

exports[`Notification > renders with close button correctly 1`] = `
"<transition-stub enteractiveclass="transform ease-out duration-300 transition" enterfromclass="translate-y-2 opacity-0 sm:translate-y-0 sm:translate-x-2" entertoclass="translate-y-0 opacity-100 sm:translate-x-0" leaveactiveclass="transition ease-in duration-100" leavefromclass="opacity-100" leavetoclass="opacity-0" appear="true" persisted="false" css="true">
  <div class="w-full pointer-events-auto bg-white dark:bg-gray-900 rounded-lg shadow-lg" role="status">
    <div class="relative overflow-hidden rounded-lg ring-1 ring-gray-200 dark:ring-gray-800">
      <div class="flex p-4 gap-3 items-center">
        <!--v-if-->
        <!--v-if-->
        <div class="w-0 flex-1">
          <!--v-if-->
          <!--v-if-->
          <!--v-if-->
        </div>
        <div class="flex items-center gap-2 flex-shrink-0 mt-0">
          <!--v-if--><button type="button" class="focus:outline-none focus-visible:outline-0 disabled:cursor-not-allowed disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 text-red-500 hover:text-red-600 disabled:text-red-500 dark:text-red-400 dark:hover:text-red-500 dark:disabled:text-red-400 underline-offset-4 hover:underline focus-visible:ring-2 focus-visible:ring-inset focus-visible:ring-red-500 dark:focus-visible:ring-red-400 inline-flex items-center" aria-label="Close"><span class="i-heroicons-x-mark-20-solid flex-shrink-0 h-5 w-5" aria-hidden="true"></span><span class="">close</span>
            <!--v-if-->
          </button>
        </div>
      </div>
      <div class="absolute bottom-0 end-0 start-0 h-1 bg-primary-500 dark:bg-primary-400" ></div>
    </div>
  </div>
</transition-stub>"
`;

exports[`Notification > renders with color correctly 1`] = `
"<transition-stub enteractiveclass="transform ease-out duration-300 transition" enterfromclass="translate-y-2 opacity-0 sm:translate-y-0 sm:translate-x-2" entertoclass="translate-y-0 opacity-100 sm:translate-x-0" leaveactiveclass="transition ease-in duration-100" leavefromclass="opacity-100" leavetoclass="opacity-0" appear="true" persisted="false" css="true">
  <div class="w-full pointer-events-auto bg-white dark:bg-gray-900 rounded-lg shadow-lg" role="status">
    <div class="relative overflow-hidden rounded-lg ring-1 ring-gray-200 dark:ring-gray-800">
      <div class="flex p-4 gap-3 items-center">
        <!--v-if-->
        <!--v-if-->
        <div class="w-0 flex-1">
          <!--v-if-->
          <!--v-if-->
          <!--v-if-->
        </div>
        <div class="flex items-center gap-2 flex-shrink-0 mt-0">
          <!--v-if--><button type="button" class="focus:outline-none focus-visible:outline-0 disabled:cursor-not-allowed disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 text-gray-500 dark:text-gray-400 hover:text-gray-700 dark:hover:text-gray-200 underline-offset-4 hover:underline focus-visible:ring-inset focus-visible:ring-2 focus-visible:ring-primary-500 dark:focus-visible:ring-primary-400 inline-flex items-center" aria-label="Close"><span class="i-heroicons-x-mark-20-solid flex-shrink-0 h-5 w-5" aria-hidden="true"></span>
            <!--v-if-->
            <!--v-if-->
          </button>
        </div>
      </div>
      <div class="absolute bottom-0 end-0 start-0 h-1 bg-red-500 dark:bg-red-400" ></div>
    </div>
  </div>
</transition-stub>"
`;

exports[`Notification > renders with description correctly 1`] = `
"<transition-stub enteractiveclass="transform ease-out duration-300 transition" enterfromclass="translate-y-2 opacity-0 sm:translate-y-0 sm:translate-x-2" entertoclass="translate-y-0 opacity-100 sm:translate-x-0" leaveactiveclass="transition ease-in duration-100" leavefromclass="opacity-100" leavetoclass="opacity-0" appear="true" persisted="false" css="true">
  <div class="w-full pointer-events-auto bg-white dark:bg-gray-900 rounded-lg shadow-lg" role="status">
    <div class="relative overflow-hidden rounded-lg ring-1 ring-gray-200 dark:ring-gray-800">
      <div class="flex p-4 gap-3 items-start">
        <!--v-if-->
        <!--v-if-->
        <div class="w-0 flex-1">
          <!--v-if-->
          <p class="mt-1 text-sm leading-4 text-gray-500 dark:text-gray-400">Notification description</p>
          <!--v-if-->
        </div>
        <div class="flex items-center gap-2 flex-shrink-0 mt-0">
          <!--v-if--><button type="button" class="focus:outline-none focus-visible:outline-0 disabled:cursor-not-allowed disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 text-gray-500 dark:text-gray-400 hover:text-gray-700 dark:hover:text-gray-200 underline-offset-4 hover:underline focus-visible:ring-inset focus-visible:ring-2 focus-visible:ring-primary-500 dark:focus-visible:ring-primary-400 inline-flex items-center" aria-label="Close"><span class="i-heroicons-x-mark-20-solid flex-shrink-0 h-5 w-5" aria-hidden="true"></span>
            <!--v-if-->
            <!--v-if-->
          </button>
        </div>
      </div>
      <div class="absolute bottom-0 end-0 start-0 h-1 bg-primary-500 dark:bg-primary-400" ></div>
    </div>
  </div>
</transition-stub>"
`;

exports[`Notification > renders with icon correctly 1`] = `
"<transition-stub enteractiveclass="transform ease-out duration-300 transition" enterfromclass="translate-y-2 opacity-0 sm:translate-y-0 sm:translate-x-2" entertoclass="translate-y-0 opacity-100 sm:translate-x-0" leaveactiveclass="transition ease-in duration-100" leavefromclass="opacity-100" leavetoclass="opacity-0" appear="true" persisted="false" css="true">
  <div class="w-full pointer-events-auto bg-white dark:bg-gray-900 rounded-lg shadow-lg" role="status">
    <div class="relative overflow-hidden rounded-lg ring-1 ring-gray-200 dark:ring-gray-800">
      <div class="flex p-4 gap-3 items-center"><span class="i-heroicons-academic-cap flex-shrink-0 w-5 h-5 text-primary-500 dark:text-primary-400"></span>
        <!--v-if-->
        <div class="w-0 flex-1">
          <!--v-if-->
          <!--v-if-->
          <!--v-if-->
        </div>
        <div class="flex items-center gap-2 flex-shrink-0 mt-0">
          <!--v-if--><button type="button" class="focus:outline-none focus-visible:outline-0 disabled:cursor-not-allowed disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 text-gray-500 dark:text-gray-400 hover:text-gray-700 dark:hover:text-gray-200 underline-offset-4 hover:underline focus-visible:ring-inset focus-visible:ring-2 focus-visible:ring-primary-500 dark:focus-visible:ring-primary-400 inline-flex items-center" aria-label="Close"><span class="i-heroicons-x-mark-20-solid flex-shrink-0 h-5 w-5" aria-hidden="true"></span>
            <!--v-if-->
            <!--v-if-->
          </button>
        </div>
      </div>
      <div class="absolute bottom-0 end-0 start-0 h-1 bg-primary-500 dark:bg-primary-400" ></div>
    </div>
  </div>
</transition-stub>"
`;

exports[`Notification > renders with title correctly 1`] = `
"<transition-stub enteractiveclass="transform ease-out duration-300 transition" enterfromclass="translate-y-2 opacity-0 sm:translate-y-0 sm:translate-x-2" entertoclass="translate-y-0 opacity-100 sm:translate-x-0" leaveactiveclass="transition ease-in duration-100" leavefromclass="opacity-100" leavetoclass="opacity-0" appear="true" persisted="false" css="true">
  <div class="w-full pointer-events-auto bg-white dark:bg-gray-900 rounded-lg shadow-lg" role="status">
    <div class="relative overflow-hidden rounded-lg ring-1 ring-gray-200 dark:ring-gray-800">
      <div class="flex p-4 gap-3 items-center">
        <!--v-if-->
        <!--v-if-->
        <div class="w-0 flex-1">
          <p class="text-sm font-medium text-gray-900 dark:text-white">Notification title</p>
          <!--v-if-->
          <!--v-if-->
        </div>
        <div class="flex items-center gap-2 flex-shrink-0 mt-0">
          <!--v-if--><button type="button" class="focus:outline-none focus-visible:outline-0 disabled:cursor-not-allowed disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 text-gray-500 dark:text-gray-400 hover:text-gray-700 dark:hover:text-gray-200 underline-offset-4 hover:underline focus-visible:ring-inset focus-visible:ring-2 focus-visible:ring-primary-500 dark:focus-visible:ring-primary-400 inline-flex items-center" aria-label="Close"><span class="i-heroicons-x-mark-20-solid flex-shrink-0 h-5 w-5" aria-hidden="true"></span>
            <!--v-if-->
            <!--v-if-->
          </button>
        </div>
      </div>
      <div class="absolute bottom-0 end-0 start-0 h-1 bg-primary-500 dark:bg-primary-400" ></div>
    </div>
  </div>
</transition-stub>"
`;
